File: Mediation/MoPubExample/Objective-C/VponMopubSample_objC/VponMopubSampleObjC/Mopub/VponMopubInterstitialViewController.m
//
//  VponMopubInterstitialViewController.m
//  VponMopubSampleObjC
//
//  Created by EricChien on 2017/6/13.
//  Copyright © 2017年 Soul. All rights reserved.
//

#import "VponMopubInterstitialViewController.h"

#import <MoPub.h>
#import "MPVponInterstitialCustomEvent.h"

@interface VponMopubInterstitialViewController () <MPInterstitialAdControllerDelegate>

@property (weak, nonatomic) IBOutlet UIButton *actionButton;

@property (strong, nonatomic) MPInterstitialAdController *mpInterstitial;

@end

@implementation VponMopubInterstitialViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.title = @"Mopub - Interstitial";
    [self actionButtonDidTouch:self.actionButton];
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
    // Dispose of any resources that can be recreated.
}

#pragma mark - Button Method

- (IBAction)actionButtonDidTouch:(UIButton *)sender {
    
    sender.enabled = NO;
    
    if (self.mpInterstitial != nil && self.mpInterstitial.ready) {
        [self.mpInterstitial showFromViewController:self];
    } else {
        #warning set ad unit id
        self.mpInterstitial = [MPInterstitialAdController interstitialAdControllerForAdUnitId:@"848bf4d03e7b4fdda02be232f8e6b4d1"];
        self.mpInterstitial.delegate = self;
        self.mpInterstitial.localExtras = @{
            @"contentURL": @"https://www.vpon.com",
            @"contentData": @{@"key1": @"MoPub", @"key2": @(1.2), @"key3": @(YES)}
        };
        [self.mpInterstitial loadAd];
    }
}

#pragma mark - MPInterstitialAdController Delegate

- (void)interstitialDidLoadAd:(MPInterstitialAdController *)interstitial {
    NSLog(@"Received interstitial ad successfully");
    self.actionButton.enabled = YES;
    [self.actionButton setTitle:@"show" forState:UIControlStateNormal];
}

- (void)interstitialDidFailToLoadAd:(MPInterstitialAdController *)interstitial {
    NSLog(@"Failed to receive interstitail");
    self.actionButton.enabled = YES;
    [self.actionButton setTitle:@"request" forState:UIControlStateNormal];
}

- (void)interstitialDidAppear:(MPInterstitialAdController *)interstitial {
    NSLog(@"Interstitial did present screen");
}

- (void)interstitialDidDisappear:(MPInterstitialAdController *)interstitial {
    NSLog(@"Interstitial did dismiss screen");
}

@end

File: Mediation/MoPubExample/Swift/VponMopubSample_swift/VponMopubSampleSwift/Mopub/VponMopubInterstitialViewController.swift
//
//  VponMopubInterstitialViewController.swift
//  VponMopubSampleSwift
//
//  Created by EricChien on 2017/6/13.
//  Copyright © 2017年 Soul. All rights reserved.
//

import UIKit

import MoPub

class VponMopubInterstitialViewController: UIViewController {
    
    @IBOutlet weak var actionButton: UIButton!
    
    var mpInterstitial : MPInterstitialAdController!
    
    override func viewDidLoad() {
        super.viewDidLoad()
        self.title = "Mopub - Interstitial"
        actionButtonDidTouch(actionButton)
    }
    
    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
    }
    
    // MARK: Button Method
    
    @IBAction func actionButtonDidTouch(_ sender: UIButton) {
        sender.isEnabled = false
        
        if mpInterstitial != nil && mpInterstitial.ready {
            mpInterstitial.show(from: self)
        } else {
            // TODO: set ad unit id
            mpInterstitial = MPInterstitialAdController(forAdUnitId: "848bf4d03e7b4fdda02be232f8e6b4d1")
            mpInterstitial.localExtras = [
                "contentURL":"https://www.vpon.com",
                "contentData": ["key1": "Mopub", "key2": 1.2, "key3": true]
            ]
            mpInterstitial.delegate = self
            mpInterstitial.loadAd()
        }
    }
}

extension VponMopubInterstitialViewController: MPInterstitialAdControllerDelegate {
    
    func interstitialDidLoadAd(_ interstitial: MPInterstitialAdController) {
        print("Received interstitial ad successfully")
        actionButton.isEnabled = true
        actionButton.setTitle("show", for: .normal)
    }
    
    func interstitialDidFail(toLoadAd interstitial: MPInterstitialAdController) {
        print("Failed to receive interstitail")
        actionButton.isEnabled = true
        actionButton.setTitle("request", for: .normal)
    }
    
    func interstitialDidAppear(_ interstitial: MPInterstitialAdController) {
        print("Interstitial did present screen")
    }
    
    func interstitialDidDisappear(_ interstitial: MPInterstitialAdController) {
        print("Interstitial did dismiss screen")
    }
}
